contents/blog/mobile-session-replay.md

```diff
@@ -0,0 +1,144 @@
+
+## The prerequisites for mobile session replay
+
+First, none of this would be possible without [Manoel](/community/profiles/30206). We had mobile experience, but not the dedicated mobile SDK experience needed for this big of a feature. Manoel had that experience and this was only possible thanks to him.
+
+When Manoel joined the first thing he worked on was [rewriting](https://github.com/PostHog/product-internal/issues/506) the SDKs in Kotlin and Swift, removing code we didn't use, improving tests, automating deployments, and making sure they worked with the latest platforms.
+
+We already had the other prerequisite, our existing session replay infrastructure. We had the data structure, the way to store replays, as well as a complete product for playback and analysis. All this could be reused for mobile replay.
+
+![Replay](https://res.cloudinary.com/dmukukwp6/image/upload/replay_03a8c56981.png)
+
+Importantly, Paul and Manoel realized the mobile data needed to be transformed into a format the rrweb player could use. They wanted to keep the rrweb schema to ensure the fewest changes possible to our API and player. To do this, they wrote a [validation and testing tool](https://github.com/PostHog/mobile-replay-data-transformer) to rapidly test the transformations before deploying it in our main app. 
+
+Once this was done, everything was ready for the mobile replay capture to be worked on. 
+
+## How we built mobile session replay
+
+Work started by developing a [proof of concept](https://github.com/PostHog/posthog-android/pull/69) for Android session replay. This developed from sending anything from an Android app and being replayed to basic components like text and images.
+
+![PoC](https://res.cloudinary.com/dmukukwp6/image/upload/mobile_b85c032c93.png)
+
+From here, there was wireframe capture along with logs and network requests. Afterwards, there were standard Android components like radios, checkboxes, Calendar, [Toggle](https://github.com/PostHog/posthog/pull/19279), RatingBar, and more. 
+
+These need to be transformed to render as an HTML wireframe (as rrweb expects). Many of them required custom transformation and components to render properly. For example, radio buttons didn't group, padding wasn't applied, and positioning was wrong.
+
+Beyond this, fitting the data into a service meant for web caused many challenges. For example, click events weren't showing even though the types and data were the same as data that worked. After some investigation, we found that rrweb expects touch events to be associated with specific elements. Setting the ID to the `body` element was enough to fix it.
+
+Once we got consistent and useful results from what we built, we recruited our first test users and iterated from there. Later, we followed a similar process for [iOS](https://github.com/PostHog/posthog-ios/pull/115).
+
+## Solving mobile replays big problems
```

Comment:
There's something off about the formatting of this section. We did such a great job outlining the problems we faced, but there isnt a 1-1 matching here (or at least the formatting breaks the rhythym)

Perhaps we turn **Multiple platforms** into its on subheading, and then in the section we include the above section on how we built it in android

e.g this bit and the rest of the section...

> Work started by developing a [proof of concept](https://github.com/PostHog/posthog-android/pull/69) for Android session replay. This developed from sending anything from an Android app and being replayed to basic components like text and images.